app/models/flight_m/copy_to_category.rb

```diff
@@ -0,0 +1,34 @@
+module FlightM
+  module CopyToCategory
+    def copy_to_category(category)
+      new_attrs = self.attributes
+      new_attrs.delete('id')
+      new_attrs.delete('category_id')
+      new_attrs['category_id'] = category.id
+      flight = Flight.find_by(contest_id: self.contest_id,
+        name: self.name, category_id: category.id)
+      if (flight)
+        flight.update!(new_attrs)
+      else
+        new_attrs['category_id'] = category.id
+        flight = Flight.create!(new_attrs)
+      end
+      copy_pilot_flights_to(flight)
+      flight
+    end
+
+    def copy_pilot_flights_to(flight)
+      new_pilots = self.pilot_flights.collect(&:pilot_id)
```

Comment:
These local variable names are mortally confusing. `self` is an existing Flight object that's being copied to the new synthetic category. `flight` is the new Flight object created in `copy_to_category` (unless it already existed because `copy_to_category` was previously invoked on the same instance of `self`). But `new_pilots` is a list of `Pilot.id` values from (old) `self`, and `prior_pilots` are taken from (new) `flight`.